tests/catalog/test_sql.py

```diff
@@ -1274,6 +1275,7 @@ def test_commit_table(catalog: SqlCatalog, table_schema_nested: Schema, table_id
     assert new_schema
     assert new_schema == update._apply()
     assert new_schema.find_field("b").field_type == IntegerType()
+    assert updated_table_metadata.last_updated_ms > last_updated_ms
```

Comment:
Could we also add similar checks to unit tests for `update_table_metadata`? For example:
https://github.com/apache/iceberg-python/blob/dceedfac4ec072ee4da99bf02dc93c1d27be45a9/tests/table/test_init.py#L597-L684
https://github.com/apache/iceberg-python/blob/dceedfac4ec072ee4da99bf02dc93c1d27be45a9/tests/table/test_init.py#L753-L761